GTMAppAuthSwift/Sources/AuthStateStore.swift

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+/// Represents the operations needed to provide `AuthState` storage.
+@objc(GTMAuthStateStore)
+public protocol AuthStateStore {
+  /// The `String` name for the stored auth state.
+  @objc var itemName: String { get }
```

Comment:
Let's consider leaving `itemName` an implementation detail.  This would significantly simplify the protocols and make them more implementation agnostic.  `KeychainStore` could opt to use the initializer below and expose this property.  We could also drop the `itemName` parameters from `OAuth2AuthStateStore` methods.

Reply:
Removed `itemName` from the protocol.